src/app/components/page-header.tsx

```diff
@@ -0,0 +1,85 @@
+import { memo } from 'react';
+import { FiArrowLeft, FiX } from 'react-icons/fi';
+import { useNavigate } from 'react-router-dom';
+
+import { Box, Flex, FlexProps, IconButton } from '@stacks/ui';
+
+import { RouteUrls } from '@shared/route-urls';
+
+import { NetworkModeBadge } from '@app/components/network-mode-badge';
+import { Title } from '@app/components/typography';
+
+interface PageHeaderProps extends FlexProps {
+  actionButton?: JSX.Element;
+  closeIcon?: boolean;
+  hideActions?: boolean;
+  onClose?(): void;
+  enableClose?: boolean;
```

Comment:
I think some of these props are inspired from @fbwoolf's work on the Ledger modals. I wonder, `enableClose` can't that be inferred from whether `onClose` is defined, same for goback logic

Reply:
I decided to rename this props to defaultClose / defaultGoBack and make them stay there in order not to call same navigation function 8 times in 4 different files